src/energyplus/ForwardTranslator/ForwardTranslateScheduleRuleset.cpp

```diff
@@ -283,11 +283,14 @@ namespace energyplus {
         if (date == dec31) {
 
           // we may now think of date as being the next saturday on or after 12/31
+          // we may need 2 more week schedules:
+          // 1 if the week that includes 12/31 is the same as the last week schedule
+          // 2 if the week that includes 12/31 is different from the last week schedule
 
           // set last week schedule before we overwrite week schedule
-          lastDate = date - Time(1);
-          while (lastDate.dayOfWeek().value() != DayOfWeek::Saturday) {
```

Comment:
I'm fairly certain that this bug is relevant for simulations whose year ends on a Sunday. The `lastDate = date - Time(1)` line here makes `lastDate` a Saturday, and so we don't enter this `while` block.